### PR TITLE
handle empty mode objects

### LINKED ISSUE #165: JS plugin inserts extra comma when `$extension.modes` is empty object
Tested using:

```json
    "@cobalt-ui/cli": "^1.6.2",
    "@cobalt-ui/plugin-js": "^1.4.2"
```

I'm seeing this in both the `meta` and `modes` top level keys.

image

image

### PR DESCRIPTION
**Pull request by dev-nicolaos**
## Changes

Fixes #165 

## How to Review

Copy only the changes from _packages/plugin-js/test/color/tokens.json_ into the main branch, run the tests, and observe that the output of the color test is invalid due to the extra commas that get inserted. Then observe that the tests on this branch pass thanks to the additional serialization logic to handle empty objects.

### CODE REVIEW

File: packages/plugin-js/src/index.ts [hunk 1]
```diff
@@ -32,6 +32,10 @@ interface TSResult {
   modes: {[id: string]: string | TSResult['modes']};
 }
 
+function isEmptyObject(obj: object): boolean {
+  return Object.keys(obj as object).length === 0;
+}
```
Comment (dev-nicolaos) [inline packages/plugin-js/src/index.ts:37]: This logic is pulled from https://github.com/drwpow/cobalt-ui/blob/main/packages/core/src/util.ts#L10. I would have used that function directly, but its not exported from the core module and I didn't know if its desired to keep that API as narrow as possible.

**Comment by dev-nicolaos** [comment]
@drwpow It looks like the tests are only failing on the Windows/Node 21 test, but the error isn't particularly helpful (just a timeout). I can't replicate it on my windows machine as `pnpm run build` fails locally (I think there's a `cp` in one of the build scripts). Any thoughts on why it might be failing?

**Review by drwpow — APPROVED** [review]
Great fix, and great tests, thank you!

**Comment by drwpow** [comment]
Yeah I’m not sure why that Windows runner timed out. I haven’t noticed it being unnecessarily slow. But will monitor and hopefully it was just a rare occurrence. Either way, unrelated to your changes.

**Pull request merged**